automation/src/main/java/org/wso2/carbon/testgrid/automation/executor/JMeterExecutor.java

```diff
@@ -0,0 +1,131 @@
+                            ", Test Location: ", testLocation, ", JMeter Home: ", jMeterHome, "}"));
+        }
+        overrideJMeterConfig(testLocation, testName, deployment); // Override JMeter properties for current deployment.
+        JMeterUtils.setJMeterHome(jMeterHome);
+        JMeterUtils.initLocale();
+        try {
+            SaveService.loadProperties();
+        } catch (IOException e) {
+            throw new TestAutomationException("Error occurred when initialising JMeter save service");
+        }
+
+        HashTree testPlanTree;
+        try {
+            testPlanTree = SaveService.loadTree(new File(script));
+        } catch (IOException e) {
+            throw new TestAutomationException("Error occurred when loading test script.", e);
+        }
+
+        Summariser summariser = null;
+        String summariserName = JMeterUtils.getPropDefault("summariser.name", "summary");
+        if (summariserName.length() > 0) {
+            summariser = new Summariser(summariserName);
+        }
+
+        Path scriptFileName = Paths.get(script).getFileName();
+        if (scriptFileName == null) {
+            throw new TestAutomationException(StringUtil.concatStrings("Script file ", script, " cannot be located."));
+        }
+        String resultFile =
+                Paths.get(testLocation, "Results", "Jmeter", scriptFileName + ".xml").toAbsolutePath().toString();
+        ResultCollector resultCollector = new ResultCollector(summariser);
+
+        resultCollector.setFilename(resultFile);
+        testPlanTree.add(testPlanTree.getArray()[0], resultCollector);
+
+        // Run JMeter Test
+        jMeterEngine.configure(testPlanTree);
+        jMeterEngine.run();
+        jMeterEngine.exit();
+    }
+
+    @Override
+    public void init(String testLocation, String testName) throws TestAutomationException {
+        jMeterHome = EnvironmentUtil.getSystemVariableValue(TestGridConstants.JMETER_HOME);
+        if (StringUtil.isStringNullOrEmpty(jMeterHome)) {
+            throw new TestAutomationException(
+                    StringUtil.concatStrings(TestGridConstants.JMETER_HOME, " environment variable not set."));
+        }
+
+        this.testName = testName;
+        this.testLocation = testLocation;
+    }
+
+    /**
+     * Overrides the JMeter properties with the properties required for the current deployment.
+     *
+     * @param testLocation directory location of the tests
+     * @param testName     test name
+     * @param deployment   deployment details of the current pattern
+     */
+    private void overrideJMeterConfig(String testLocation, String testName, Deployment deployment) {
+        Path path = Paths.get(testLocation, "JMeter", testName, "src", "test", "resources", "user.properties");
+        if (Files.exists(path)) {
+            JMeterUtils.loadJMeterProperties(path.toAbsolutePath().toString());
```

Comment:
A utils class should never store state. Is it the case here?

Reply:
True. But this is the JMeters way of loading JMeter properties (required) and hence we have to follow it.